app/lib/exchange-rate/money.ts

```diff
@@ -0,0 +1,458 @@
+   *
+   * Note that dividing a monetary amount cannot be exact in all cases.
+   * E.g. 10 USD / 3 = 3.33 USD
+   *
+   * The division is performed with a precision of 20 decimals before
+   * rounding back to the monetary amount. (See https://mikemcl.github.io/big.js/#dp)
+   */
+  divide = (divisor: NumberInput): Money => {
+    const amount = this._data.amount.div(divisor);
+    return this.merge({ amount });
+  };
+
+  add = (money: Money): Money => {
+    this.assertSameCurrency(money);
+    const amountInMinUnit = money.amount(this._data.unit.name);
+    return this.merge({ amount: this._data.amount.plus(amountInMinUnit) });
+  };
+
+  subtract = (money: Money): Money => {
+    this.assertSameCurrency(money);
+    const amountInMinUnit = money.amount(this._data.unit.name);
+    return this.merge({ amount: this._data.amount.minus(amountInMinUnit) });
+  };
+
+  abs = (): Money => {
+    return this.merge({ amount: this._data.amount.abs() });
+  };
+
+  equals = (money: Money): boolean => {
+    return (
+      this.currency === money.currency &&
+      this._data.amount.eq(money.amount(this._data.unit.name))
+    );
+  };
+
+  greaterThan = (money: Money): boolean => {
+    this.assertSameCurrency(money);
+    const amountInMinUnit = money.amount(this._data.unit.name);
+    return this._data.amount.gt(amountInMinUnit);
+  };
+
+  greaterThanOrEqual = (money: Money): boolean => {
+    this.assertSameCurrency(money);
+    const amountInMinUnit = money.amount(this._data.unit.name);
+    return this._data.amount.gte(amountInMinUnit);
+  };
+
+  lessThan = (money: Money): boolean => {
+    this.assertSameCurrency(money);
+    const amountInMinUnit = money.amount(this._data.unit.name);
+    return this._data.amount.lt(amountInMinUnit);
+  };
+
+  lessThanOrEqual = (money: Money): boolean => {
+    this.assertSameCurrency(money);
+    const amountInMinUnit = money.amount(this._data.unit.name);
+    return this._data.amount.lte(amountInMinUnit);
+  };
+
+  isZero = (): boolean => {
+    return this._data.amount.eq(0);
+  };
+
+  isPositive = (): boolean => {
+    return this._data.amount.gt(0);
+  };
+
+  isNegative = (): boolean => {
+    return this._data.amount.lt(0);
+  };
+
+  /**
+   * Converts money to a regular javascript number for the requested currency unit.
+   * If no unit is provided, base/default unit is used (bitcoin for BTC, dollar for USD, etc.)
+   * Throws an error if it's not possible to keep full precision.
+   */
+  toNumber = (unit?: string): number => {
+    // Don't use big.js toNumber because it sometimes returns -0.
+    const str = this.toString(unit);
+    const num = Number(str);
+
+    if (
+      str !==
+      this.merge({
+        amount: num,
+        unit: unit ?? this.currencyData.baseUnit,
+      }).toString(unit)
+    ) {
+      throw new Error('Converting to number was imprecise');
+    }
+
+    return num;
+  };
+
+  /**
+   * Returns the money value as a stringified number without a currency.
```

Comment:
I followed the approach from that lib but perhaps we should print it with currency? or have an option to do it with or without like in the toLocaleString method below

Reply:
The only reason I see would be if we use `toString` to show some value in the UI, but in that case using `toLocaleString` would make more sense to get the additional formatting

Reply:
toString is also called automatically when you try to use the money in string interpolation for example. so:
```
x = new Money(...)
console.log(`Some message ${x}`)
```

Reply:
So if we wanted the string interpolation to show the symbol we would have to make that the default? 
Might as well add a `showCurrency` flag and have `true` be the default.

Should we then  have a `fromString` method?

Reply:
> Should we then have a fromString method?

what for?

Reply:
Nothin in particular, I was just thinking if we are converting to a string and including the symbol then we would be able to create a new Money object from that string, but I don't have a use case in mind

Reply:
we can add later if needed

Reply:
resolving this for now. we can keep as is and add later if needed